src/main/java/uk/gov/mca/beacons/api/services/LegacyBeaconService.java

```diff
@@ -42,11 +65,35 @@ private void validate(LegacyBeacon beacon) {
     }
   }
 
-  public void deleteAll() {
-    legacyBeaconGateway.deleteAll();
+  public Optional<List<LegacyBeacon>> findMatchingLegacyBeacons(Beacon beacon) {
+    if (
+      beacon.getHexId() == null || beacon.getAccountHolderId() == null
+    ) return Optional.empty();
```

Comment:
Update tests that use `CreateRegistration` so that `Beacon` has a hexId and accountHolderId.

Reply:
Decided to push back due to effort / reward ratio.  Added TODO.